crates/lib3h/src/transport/websocket/actor.rs

```diff
@@ -0,0 +1,442 @@
+                                    }
+                                    Err(error) => {
+                                        trace!(
+                                            "Could not connect to {}! Transport error: {:?}",
+                                            uri.to_string(),
+                                            error
+                                        );
+                                    }
+                                }
+                            }
+
+                            // If we have a connection now, send payload:
+                            if let Some(connection_id) = maybe_connection_id {
+                                trace!(
+                                    "(GhostTransportWebsocket).SendMessage from {} to  {} | {:?}",
+                                    my_addr,
+                                    uri,
+                                    payload
+                                );
+                                // Send it data from us
+                                match self.streams.send(&[&connection_id], &payload.as_bytes()) {
+                                    Ok(()) => {
+                                        msg.respond(Ok(RequestToChildResponse::SendMessageSuccess))?
+                                    }
+                                    Err(error) => msg.respond(Err(error))?,
+                                }
+                            } else {
+                                msg.respond(Err(TransportError::new(format!(
+                                    "Unable to acquire connection. Can't send to {}",
+                                    uri,
+                                ))))?;
+                            }
+                        }
+                    };
+                }
+            }
+        }
+
+        // make sure we have bound and get our address if so
+        let my_addr = match &self.bound_url {
+            Some(my_addr) => my_addr.clone(),
+            None => return Ok(false.into()),
+        };
+
+        trace!("Processing for: {}", my_addr);
+
+        let (did_work, stream_events) = self.streams.process()?;
+        for event in stream_events {
+            match event {
+                StreamEvent::ErrorOccured(connection_id, error) => {
+                    let uri = self
+                        .streams
+                        .connection_id_to_url(connection_id)
+                        .unwrap_or_else(|| {
+                            Url::from_str("wss://0.0.0.0")
+                                .expect("URL literal is syntactically correct")
+                        });
+                    self.endpoint_self.publish(
+                        Lib3hSpan::todo(),
+                        RequestToParent::ErrorOccured { uri, error },
+                    )?;
+                }
+                StreamEvent::ConnectResult(_connection_id, _) => {}
```

Comment:
When the user of this lib sends in a SendMessage, and we are not already connected to that remote URI, the ghost actor contract is for the SendMessageResult response to only be sent out after all the connection handshaking is complete and the actual low-level send is invoked.

This is an awesome iteration, I'm totally cool with putting a comment to this effect here and iterating as a follow-on.